lib/molinillo/resolution.rb

```diff
@@ -320,12 +320,24 @@ def attempt_to_activate_existing_spec(existing_node)
           new_requirements = requirements.dup
           push_state_for_requirements(new_requirements)
         else
+          return if attempt_to_swap_possibility
           create_conflict
           debug(depth) { "Unsatisfied by existing spec (#{existing_node.payload})" }
           unwind_for_conflict
         end
       end
 
+      # Attempts to swp the current {#possibility} with the already-activated
+      # spec with the given name
+      # @return [Boolean] Whether the possibility was swapped into {#activated}
+      def attempt_to_swap_possibility
+        swapped = activated.dup
+        swapped.vertex_named(name).payload = possibility
+        return unless swapped.vertex_named(name).requirements.
```

Comment:
Nil is a Boolean value in Ruby 

-Samuel E. Giddins

On Feb 14, 2015, at 7:38 AM, Kyle Fuller <notifications@github.com> wrote:

In lib/molinillo/resolution.rb:

>            create_conflict
>            debug(depth) { "Unsatisfied by existing spec (#{existing_node.payload})" }
>            unwind_for_conflict
>          end
>        end
>  
> +      # Attempts to swp the current {#possibility} with the already-activated
> +      # spec with the given name
> +      # @return [Boolean] Whether the possibility was swapped into {#activated}
> +      def attempt_to_swap_possibility
> +        swapped = activated.dup
> +        swapped.vertex_named(name).payload = possibility
> +        return unless swapped.vertex_named(name).requirements.
The method documentation states this returns boolean but isn't the case here?

—
Reply to this email directly or view it on GitHub.